### PR TITLE
Adding "Myself" page for status checker in public

### PR DESCRIPTION
**Pull request by nibivi77**
### Description
Adding "Myself" page for status checker in public

### Related Azure Boards Work Items
[AB#3546](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3546)

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### Test Instructions
- Navigate to `http://localhost:3000/en/status`
- Select Myself
- Enter: 694913 and 903002863

### Additional Notes
image
Should the hcaptcha references be removed in the myself page?

### CODE REVIEW

File: frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+                placeholder={t('status:myself.form.application-code-placeholder')}
+                required
+                errorMessage={errorMessages.code}
+              />
+              <InputField id="sin" name="sin" label={t('status:myself.form.sin-label')} helpMessagePrimary={t('status:myself.form.sin-description')} required errorMessage={errorMessages.sin} />
+              <Button variant="primary" id="submit" disabled={isSubmitting} className="my-8" data-gc-analytics-formsubmit="submit">
+                {t('status:myself.form.submit')}
+                <FontAwesomeIcon icon={isSubmitting ? faSpinner : faChevronRight} className={cn('ms-3 block size-4', isSubmitting && 'animate-spin')} />
+              </Button>
+            </fetcher.Form>
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx:222]: - I feel like there should be some spacing between the following three  "Complete all fields" "Application Code or Client Number", and "Social Insurance Number (SIN)":

- I feel that the input for "Application Code or Client Number" should be extended as the placeholder text cannot be seen

Mobile view screenshot:
image

**Review by nicholas-ly — COMMENTED** [review]
That "Only one captcha is permitted per parent container" is a result of [React's StrictMode](https://react.dev/reference/react/StrictMode#fixing-bugs-found-by-double-rendering-in-development:~:text=To%20help%20you%20find%20accidentally%20impure%20code%2C%20Strict%20Mode%20calls%20some%20of%20your%20functions%20(only%20the%20ones%20that%20should%20be%20pure)%20twice%20in%20development). I don't know how to get around this message in development.

You won't see this message if you run the application in production mode.

File: frontend/app/routes/$lang+/_public+/status+/index.tsx [hunk 2]
```diff
@@ -76,10 +76,11 @@ export async function action({ context: { session }, params, request }: ActionFu
 
   if (statusCheckFor === CheckFor.Myself) {
     //TODO: redirect to $lang+/_public+/status+/myself
+    return redirect(getPathById('$lang+/_public+/status+/myself+/index', params));
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_public+/status+/index.tsx:79]: Can the `TODO` be removed?

File: frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+              <FontAwesomeIcon icon={isSubmitting ? faSpinner : faChevronRight} className={cn('ms-3 block size-4', isSubmitting && 'animate-spin')} />
+            </ButtonLink>
+          </>
+        ) : (
+          <>
+            {errorSummaryItems.length > 0 && <ErrorSummary id={errorSummaryId} errors={errorSummaryItems} />}
+            <fetcher.Form method="post" onSubmit={handleSubmit} noValidate autoComplete="off" data-gc-analytics-formname="ESDC-EDSC: Canadian Dental Care Plan Status Checker">
+              <input type="hidden" name="_csrf" value={csrfToken} />
+              {hCaptchaEnabled && <HCaptcha size="invisible" sitekey={siteKey} ref={captchaRef} />}
+              <p className="italic">{t('status:myself.form.complete-fields')}</p>
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx:207]: Move the `Complete all fields.` paragraph outside the form and before the `ErrorSummary`.

https://www.w3.org/WAI/tutorials/forms/instructions/
> Provide such instructions before the `form` element to ensure that it is read aloud by screen readers before they switch to “Forms Mode”.

File: frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+              <InputField
+                id="code"
+                name="code"
+                label={t('status:myself.form.application-code-label')}
+                helpMessagePrimary={t('status:myself.form.application-code-description')}
+                placeholder={t('status:myself.form.application-code-placeholder')}
+                required
+                errorMessage={errorMessages.code}
+              />
+              <InputField id="sin" name="sin" label={t('status:myself.form.sin-label')} helpMessagePrimary={t('status:myself.form.sin-description')} required errorMessage={errorMessages.sin} />
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx:217]: Wrap those two input fields with a `div` and the following classes `mb-8 space-y-6` to be consistent with other pages.

File: frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+              <Button variant="primary" id="submit" disabled={isSubmitting} className="my-8" data-gc-analytics-formsubmit="submit">
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx:218]: Remove the `my-8` class name.

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

File: frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx (hunk 1 above)
Comment (nibivi77) [reply frontend/app/routes/$lang+/_public+/status+/myself+/index.tsx:222]: I would remove this placeholder.

**Review by sebastien-comeau — DISMISSED** [review]

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**